freelist.go

```diff
@@ -6,18 +6,29 @@ import (
 	"unsafe"
 )
 
+
+// txPending holds a list of pgids and corresponding allocation txns
+// that are pending to be freed.
+type txPending struct {
+	ids              []pgid
+	alloctx          []txid // txids allocating the ids
```

Comment:
1, open read txns txid=7,8,9
2. allocate p at txid=10; close txn
3. open read txns txid=11,12; both depend on p
4. free p at txid=13; close txn

Closing txid=13 can't free p until 11 and 12 are closed, but it doesn't have to wait for 7,8,9 to close since the minimum txn (tracked by alloctx) that can possibly use the page is txid=10.